services/spar/src/Spar/Intra/Brig.hs

```diff
@@ -258,6 +269,20 @@ setBrigUserRichInfo buid richInfo = do
      | otherwise
        -> throwSpar . SparBrigError . cs $ "set richInfo failed with status " <> show sCode
 
+getBrigUserRichInfo :: (HasCallStack, MonadSparToBrig m) => UserId -> m (Maybe RichInfo)
+getBrigUserRichInfo buid = do
+  resp <- call
+    $ method GET
+    . paths [ "users", toByteString' buid, "rich-info" ]
+    . header "Z-User" (toByteString' buid)
+    . header "Z-Connection" ""
+  case statusCode resp of
+    200 -> do
+      Just <$> parseResponse @RichInfo resp
+    404   -> pure Nothing
```

Comment:
I think `brig` should never return a `404` -> unless the user is not found which would be some kind of DB inconsistency? If there's no rich info what is returned is a 200 with:

```
{"version":0,"fields":[]}
```

Reply:
Code doesn't agree with you. It returns a 404 if the user isn't found from what I could see

Reply:
Yes; my point was more in the lines of: how did u get a userId on the SCIM to a user that does not exist? It exists on the SCIM table but not on brig? This sounds like a very inconsistent state, don't u agree?

Reply:
What if the brig user is deleted from the brig side inbetween? This is a usecase we apparently support; given we have other tests that assert this behaviour (see `deleteUserOnBrig` calls in the test-suite).   However we use the `deleted` flag on `brigUser` to decide if the user is present or not.   I don't actually know what the `deleted` flag means (does it mean it is 'marked' for deletion; or do we never delete user data?) If it's the first; then I think this function is correct as is; if it is the second then erroring out might be a sane thing to do; though I still think it doesn't harm us to silently ignore this?





(grml; i miss transactions)

Reply:
I don't have enough context to answer all your questions, but I can answer this one:

>  I don't actually know what the deleted flag means (does it mean it is 'marked' for deletion; or do we never delete user data?)

Upon deletion of a user in brig, all user data is deleted (except the user's UUID in the `user` table), and a so-called `tombstone` is inserted: https://github.com/wireapp/wire-server/blob/develop/services/brig/src/Brig/API/User.hs#L788-L789 - i.e. the only thing remaining in the database is the UUID and a flag "deleted", to differentiate a now-deleted user from a never-existed user.